### ISSUE TITLE
[Question] About rate limits

### ISSUE THREAD

**Issue by Suleman-Elahi** [zonca/twitter_blocklist#4]
How many accounts can I block with it?

I have over a million Twitter accounts ..... usually the trolls and fakes(in CSV) and I want to block them all. Will this tool work? If not then can you explain how much time will it take to block all those accounts considering the API limits if I use your tool. Thanks.

**Comment by zonca** [comment]
It should work. About 300k in 1 hour.

**Comment by Suleman-Elahi** [comment]
Thanks for your reply..... that seems quite long time. Do you know of any faster way of doing that?

**Comment by zonca** [comment]
Just leave it running overnight

**Comment by EpicOrange** [comment]
Hey, it's 2021 and this still works! But my API keys got revoked after around 200 blocks, for violating some unspecified Terms etc etc. As expected, no response from support.

Any idea what the block rate limits are actually? I couldn't find them myself.

**Comment by EpicOrange** [comment]
Oh.. [I found them](https://help.twitter.com/en/rules-and-policies/twitter-automation), but you may not like them...

> **Automated adding to lists or collections**:
> You may not add Twitter users to lists or add Tweets to collections in a bulk or indiscriminate manner. Adding a large number of unrelated users to lists is a violation of the Twitter Rules.

this sucks

**Comment by zonca** [comment]
not sure, are "blocks" considered collections?

**Comment by Suleman-Elahi** [comment]
@zonca but "blocks" is a kind of list ...no ?

**Comment by zonca** [comment]
there in the text it seems they refer to the actual Twitter Lists

@Suleman-Elahi have you also had keys revoked?

**Comment by Suleman-Elahi** [comment]
I am yet to complete my app. And now I am a little scared xD ... maybe you are right... because Block Together wasn't shut down for abusing the API or violating Twitter terms. https://blocktogether.org/

**Comment by SAUUU** [comment]
my list is 1800 accounts to block only 
but whenever i start the script,  the App API get revoked from twitter

**Comment by zonca** [comment]
Are you loading from a CSV or it is a Twitter list?

**Comment by SAUUU** [comment]
> Are you loading from a CSV or it is a Twitter list?

CSV i exported from the same script from my other account, 
always the app gets shut down for violating Twitter terms at 50~ account block  progress

**Comment by zonca** [comment]
I just tested now importing 2000 blocks into my test account and worked fine.

do you mind attaching your blocklist here on github so I can test with yours? or maybe just the first 100 or so.

**Comment by SAUUU** [comment]
sure here it is

[my_blocks.zip](https://github.com/zonca/twitter_blocklist/files/5928708/my_blocks.zip)

**Comment by zonca** [comment]
it works fine on my account, I imported them all without errors.
maybe something with your developer account?

**Comment by SAUUU** [comment]
will you help me, if I signed to your API and you upload my blocklist?

**Comment by zonca** [comment]
@SAUUU I am building a web app to do this so you don't have to register a twitter app, it is not ready yet,
but I have an early prototype you can try.

* Go to https://twitter-blocklist-auth.glitch.me/
* login with Twitter
* paste the url of your blocklist, I uploaded it for you at: https://pastebin.com/raw/v6YgBRgw

@Suleman-Elahi if you want to test it as well it would be useful, check the docs at https://github.com/zonca/twitter_blocklist/blob/master/run_twitter_blocklist.ipynb on how to upload a csv list to pastebin or gist

**Comment by zonca** [comment]
the source of the website is public: https://glitch.com/edit/#!/twitter-blocklist-auth

**Comment by ilovefreesw** [comment]
> It should work. About 300k in 1 hour.

How did you reach this number? The docs say 50 requests to block endpoint in 15-minute window.

**Comment by zonca** [comment]
@ilovefreesw yes, looking at https://developer.twitter.com/en/docs/twitter-api/rate-limits#v2-limits I agree with you.
However, when I was testing with 2000 blocks, it was taking a few minutes, definitely not 10 hours.

So maybe I was using an older version of the API? I havent used this in a long while.

**Comment by ilovefreesw** [comment]
> So maybe I was using an older version of the API? I havent used this in a long while.

Fine.
